File: go/2099-find-subsequence-of-length-k-with-the-largest-sum.go
import (
    "sort"
)

func maxSubsequence(nums []int, k int) []int {
    type Pair struct {
        val int
        idx int
    }

    valueWithIndex := make([]Pair, len(nums))
    for i, num := range nums {
        valueWithIndex[i] = Pair{val: num, idx: i}
    }

    sort.Slice(valueWithIndex, func(i, j int) bool {
        return valueWithIndex[i].val > valueWithIndex[j].val
    })

    topK := valueWithIndex[:k]

    sort.Slice(topK, func(i, j int) bool {
        return topK[i].idx < topK[j].idx
    })

    result := make([]int, k)
    for i, p := range topK {
        result[i] = p.val
    }
    return result
}

File: python/2099-find-subsequence-of-length-k-with-the-largest-sum.py
class Solution:
    def maxSubsequence(self, nums, k):
        value_with_index = [(num, i) for i, num in enumerate(nums)]
        value_with_index.sort(key=lambda x: -x[0])
        top_k = sorted(value_with_index[:k], key=lambda x: x[1])
        return [num for num, _ in top_k]
